code/game/mecha/equipment/tools/mining_tools.dm

```diff
@@ -13,9 +11,9 @@
 /obj/item/mecha_parts/mecha_equipment/drill/action(atom/target)
 	if(!action_checks(target))
 		return
-	if(istype(target, /turf) && !istype(target, /turf/simulated))
-		return
-	if(isobj(target))
+	if(istype(target, /turf) && !istype(target, /turf))
```

Comment:
look at what your checking is true here. then look at what your checking is false. then slap yourself in the face for not proofing your code.